packages/flutter/lib/src/material/date_picker_theme.dart

```diff
@@ -944,6 +975,9 @@ class _DatePickerDefaultsM2 extends DatePickerThemeData {
         elevation: 24.0,
         shape: const RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(4.0))),
         dayShape: const MaterialStatePropertyAll<OutlinedBorder>(CircleBorder()),
+        yearShape: const MaterialStatePropertyAll<OutlinedBorder>(
+          RoundedRectangleBorder(borderRadius: BorderRadius.all(Radius.circular(28.0))),
+        ),
```

Comment:
Also if the comment above is true, can you add a test case to verify that the default year shape is kept with a different `textScaleFactor` (likely by verifying the `drawRRect` operation)?

Reply:
Sure.

Reply:
@dkwingsmt I have pushed the change. Since i have made it `StadiumBorder`. It's respecting border radius based on that. Also as you noted, I have written test for that.